### ISSUE TITLE
Modernize frontend

### ISSUE THREAD

**Issue by guoyunhe** [WeblateOrg/weblate#2916]
**Is your feature request related to a problem? Please describe.**
We have a Weblate instance at https://l10n.opensuse.org/ and its UI is based on Bootstrap 3.x. However, other openSUSE project websites are using a theme based on Bootstrap 4.x. If we deploy Bootstrap 4.x, there will be incompatible components/styles that may break the UI functionality.

**Describe the solution you'd like**
I hope Weblate can support Bootstrap 4.x.

**Describe alternatives you've considered**

**Additional context**

**Comment by nijel** [comment]
This has been already discussed in https://github.com/WeblateOrg/weblate/issues/1935, the biggest problem with Boostrap 4 is lack of RTL support. There were some third party mods which none of these looked complete, but I didn't investigate that for some time.

Even if that is handled, there is non trivial amount of work to migrate the UI between 3.x and 4.x.

**Comment by guoyunhe** [comment]
Yes, here are a lot of class names changed. You have to check every component carefully. If one day you decide to open a `bootstrap4` branch, I can help with the migration.

**Comment by nijel** [comment]
First we need working RTL support, then we can consider a migration. Making RTL work properly with Bootstrap 3 was painful as well, but we already have that running.

**Comment by Bachstelze** [comment]
Bootstrap 4.4 is going to be a [LTS release](https://blog.getbootstrap.com/2019/07/24/lts-plan/) and could then support [RTL](https://github.com/twbs/bootstrap/issues/19545).

**Comment by nijel** [comment]
There is also https://github.com/twbs/bootstrap/issues/28797 and https://github.com/twbs/bootstrap/issues/24807, but none really seems to be going anywhere close to having RTL supported.

**Comment by Ahoba02** [comment]
I have created a tool to solve the issue.
Please visit
http://wahamira.net/tools/get-bootstrap-rtl

And kindly give me your feedback.

**Comment by Ahoba02** [comment]
Did you check
http://wahamira.net/tools/get-bootstrap-rtl

**Comment by nijel** [comment]
Boostrap 5 should have native RTL support (see https://github.com/twbs/bootstrap/pull/30980), then it makes more sense to switch to that (once it is released).

**Comment by Bachstelze** [comment]
The beta version of Bootstrap 5 is released. While keeping jQuery as extra dependency most of our plugins should still work. Which changes will we have to figure out?

**Comment by nijel** [comment]
As Bootstrap 4 entirely changed the layout system (see https://getbootstrap.com/docs/4.6/migration/), all pages will have to be reviewed whether they still work correctly. The 5 renamed many of the helper classes, see https://getbootstrap.com/docs/5.0/migration/

**Comment by Bachstelze** [comment]
The migration looks like a lot of handwriting which we could reduce with some automatation like https://gist.github.com/landsman/526d88db01cf5ec50fad257fe2d44574
Should we start now a new fork or are there conflicts with e.g. https://github.com/WeblateOrg/weblate/issues/3063 https://github.com/WeblateOrg/weblate/issues/3558 ?

**Comment by nijel** [comment]
I don't think these issues should be blockers. In case somebody starts working on these after starting effort to migrate to Bootstrap 5, it should be built on top of that...

**Comment by Bachstelze** [comment]
Are we using dropdowns with the dropped flip option? https://getbootstrap.com/docs/5.0/migration/#dropdowns
We have many dropped "form-group", "media", "pre-scrollable", "form-inline", "arrow" and "card" styles which should be exchanged one by one.

The curret migration script is https://gist.github.com/Bachstelze/b6e83881f0608816ba444b6394eafa58

**Comment by nijel** [comment]
Yes, it's used at least on the watching dropdown menu.

**Comment by z3r0101** [comment]
Are there any plans to migrate to Bootstrap 5? Many issues present in the older version have been addressed in the latest release. Our OICT organization has flagged Bootstrap v3.4.1 as a high-security risk due to its End-of-Life status, which prevents its use in our production environment. We thought it important to inform you, as this might also impact other organizations.

**Comment by nijel** [comment]
Yes, the plan is that @meel-hd will start working on that after we are done with some cleanups in the client code (like https://github.com/WeblateOrg/weblate/issues/12172).

**Comment by nijel** [comment]
The work in progress pull request is https://github.com/WeblateOrg/weblate/pull/15113, feedback welcome.

**Comment by nijel** [comment]
Let's close this. There are still some issues, but most of the work is done. Thanks, @KarenKonou!

**Comment by Bachstelze** [comment]
Just out of curiosity: Was my migration script helpful?
